OpenScienceJournal/whistlepunk_library/src/main/java/com/google/android/apps/forscience/whistlepunk/sensors/MkrSciBleSensor.java

```diff
@@ -46,7 +45,7 @@
     private ValueHandler mValueHandler;
 
     public MkrSciBleSensor(String sensorId, MkrSciBleSensorSpec spec) {
-        super(sensorId, AppSingleton.getUiThreadExecutor());
+        super(sensorId);
```

Comment:
Can you explain the threading change here?

Reply:
It was redundant. MkrSciBleSensor extends ScalarSensor, which has two public constructors:

```
    public ScalarSensor(String id) {
        this(id, AppSingleton.getUiThreadExecutor());
    }

    @VisibleForTesting
    public ScalarSensor(String id, Executor uiThreadExecutor) {
        this(id, ExternalAxisController.DEFAULT_GRAPH_RANGE_IN_MILLIS, uiThreadExecutor,
                DEFAULT_ZOOM_LEVEL_BETWEEN_TIERS, new UptimeClock());
    }
```

Before this change, my code was calling the second one, that is marked as "VisibleForTesting". Thus I moved to the first one, which by the way internally calls the second one with `AppSingleton.getUiThreadExecutor()` as a second argument, so it's the same as before.

Reply:
Thanks! That makes sense.